static/js/publisher/metrics/graphs/activeDevices.js

```diff
@@ -1,200 +1,253 @@
 
-  const activeDevicesMetrics = bb.generate({
-    bindto: '#active_devices',
-    legend: {
-      hide: true
-    },
-    padding: PADDING,
-    tooltip: {
-      contents: snapcraftGraphTooltip.bind(this, {
-        colors: colors,
-        showLabels: true
-      }, el),
-      position: positionTooltip.bind(this, el)
-    },
-    transition: {
-      duration: 0
-    },
-    point: {
-      focus: false,
-      show: false
-    },
-    axis: {
-      x: {
-        tick: {
-          culling: false,
-          outer: true,
-          format: formatXAxisTickLabels
-        }
-      },
-      y: {
-        tick: {
-          format: formatYAxisTickLabels
-        },
-        padding: {
-          top: 30, bottom: 30
-        },
-        min: minYAxisValue
-      }
-    },
-    resize: {
-      auto: false
-    },
-    spline: {
-      interpolation: {
-        type: 'basis'
-      }
-    },
-    data: {
-      colors: colors,
-      types: types,
-      groups: [
-        group
-      ],
-      x: 'x',
-      columns: [days].concat(deviceData)
-    },
-    onrendered: () => {
-      // Because we're modifying the axis with CSS the mask crops the labels
-      // We don't need the masks, so here we remove them each render.
-      Array.from(
-        el.querySelectorAll('clipPath')
-      ).forEach(mask => {
-        if (mask.id.indexOf('axis') !== -1) {
-          mask.remove();
-        }
-      });
-    }
-  });
+  cullYAxis();
 
-  showGraph(el);
+  const tooltip = select(holder)
+    .append('div')
+    .attr('class', 'p-tooltip');
 
-  deviceData.forEach(device => {
-    const labelClass = device[0].replace(/\W+/g, '-');
-    const area = el.querySelector(`.bb-area-${labelClass}`);
-    if (area) {
-      area.dataset.label = device[0];
-    }
-  });
+  function mouseMove(d) {
+    const mousePosition = mouse(this);
 
-  // Extra events
-  let elWidth = el.clientWidth;
-
-  const resize = debounce(function () {
-    if (el.clientWidth !== elWidth) {
-      el.style.opacity = 0;
-      debounce(function () {
-        activeDevicesMetrics.resize();
-        showGraph(el);
-        elWidth = el.clientWidth;
-      }, 100)();
-    }
-  }, 500);
+    const x0 = xScale.invert(mousePosition[0]);
+    const _date = new Date(x0);
+    _date.setHours(0);
+    _date.setMinutes(0);
+    _date.setSeconds(0);
+    _date.setMilliseconds(0);
+    const i = bisectDate(_data, _date.getTime(), 0);
 
-  window.addEventListener('resize', resize);
+    const current = d.key;
+    const dateData = _data[i];
+
+    tooltip.html(tooltipTemplate(dateData, current, colorScale))
+      .style('top', `${mousePosition[1]}px`)
+      .style('left', `${mousePosition[0] + margin.left}px`)
+      .style('display', 'block');
+  }
+
+  function cancelTooltip() {
+    tooltip.style('display', 'none');
+  }
+
+  showGraph(holder);
+}
+
+export default function activeDevices(activeDevices) {
+  drawGraph(activeDevices);
+
+  select(window).on('resize', () => {
```

Comment:
Would be good to debounce resize event to prevent redrawing while user moves the window. Previous version was doing this.

Reply:
:+1: